cmd/icingadb/main.go

```diff
@@ -253,3 +258,42 @@ func run() int {
 		cancelHactx()
 	}
 }
+
+// checkDbSchema asserts the database schema of the expected version being present.
+func checkDbSchema(ctx context.Context, db *icingadb.DB, dbName string) error {
+	{
+		var one []struct{ One uint8 }
+
+		err := db.SelectContext(
+			ctx, &one,
+			"SELECT 1 as one FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA=? AND TABLE_NAME='icingadb_schema'",
+			dbName,
+		)
+		if err != nil {
+			return errors.Wrap(err, "can't check whether database schema is present")
+		}
+
+		if len(one) < 1 {
+			return errors.New("database schema is absent")
+		}
+	}
+
+	var version []struct{ Version uint16 }
+
+	err := db.SelectContext(
+		ctx, &version, "SELECT version FROM icingadb_schema ORDER BY `timestamp` DESC LIMIT 1",
```

Comment:
Please order by `id`. I wouldn't expect `timestamp` to be correct 100 percent of the time.